core/src/multiplexing/muxer.cc

```diff
@@ -479,3 +480,34 @@ void muxer::_push_to_queue(misc::shared_ptr<io::data> const& event) {
 std::string muxer::_queue_file() const {
   return (queue_file(_name));
 }
+
+/**
+ *  Remove all the queue files attached to this muxer.
+ */
+void muxer::remove_queue_file() {
```

Comment:
According to me, remove_queue_file() should not remove files by itself. It should instead call a method of persistent_file.